src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -163,15 +290,30 @@ void Instance::HandleSetCookingParameters(HandlerContext & ctx, const Commands::
     uint8_t reqCookMode;
     uint32_t reqCookTime;
     uint8_t reqPowerSetting;
-    auto & cookMode     = req.cookMode;
-    auto & cookTime     = req.cookTime;
-    auto & powerSetting = req.powerSetting;
+    uint8_t reqWattSettingIndex;
+    uint8_t maxPower = kDefaultMaxPower;
```

Comment:
these min/max power seem to only be required inside some hasfeature blocks. Please move them to have minimal scope rather than top level.

Overall this applies to all variables: please minimize their scope rather than having them all at the top level. This function has 14 top level variables and 2 arguments ... that seems high complexity for maintenance.